src/network/impl/router_impl.cpp

```diff
@@ -61,10 +55,10 @@ namespace libp2p::network {
       return Error::NO_HANDLER_FOUND;
     }
 
-    const auto &pred_hand = matched_proto.value();
-    if (matched_proto.key() == p || pred_hand.predicate(p)) {
+    const auto &[predicate, cb] = matched_proto.value();
+    if (matched_proto.key() == p or (predicate and predicate(p))) {
```

Comment:
Here, too, please assign this expression to a variable, I believe it would improve readability and debugging experience.